tests/_utils.py

```diff
@@ -0,0 +1,275 @@
+"""Utilities for testing"""
+
+from __future__ import annotations
+
+import json
+from datetime import datetime, timedelta
+from pathlib import Path
+from warnings import warn
+
+import nibabel as nib
+import numpy as np
+from nibabel import Nifti1Image
+
+CONFIG_TYPE = dict[str, dict[str, list[str | int] | int]]
+
+
+def _default_config() -> CONFIG_TYPE:
+    """Return a config."""
+    return {
+        # list of possible 'datatypes'
+        "fmap": {},
+        "func": {
+            "suffix": ["bold", "events"],
+            "tasks": ["main", "rest"],
+            "runs": [2, 1],  # nb runs for each task
+        },
+        "anat": {
+            "suffix": ["t1w", "t2w"],
+            "runs": [2, 1],  # nb runs for each suffix
+        },
+        "dwi": {"suffix": ["dwi"]},
+        # other config
+        "subject_folder_prefix": "sub-",
+        "session_folder_prefix": "ses-",
+        "timestamp_format": "%Y%m%d_%H%M%S",
+        "default_nifti_ext": ".nii.gz",
+        "layout": "flat",  # flat or nested
+        "filename_template": "$subject_$suffix_$task_$run_$timestamp",
+    }
+
+
+def create_fake_source_dataset(
+    output_dir: Path = Path.cwd() / "sourcedata",
+    subjects: str | int | list[str | int] = None,
+    sessions: None | str | int | list[str | int | None] = None,
+    datatypes: str | list[str] = None,
+    config: CONFIG_TYPE | None = None,
+) -> None:
+    """Create a fake nifti dataset.
+
+    The layout of the output dataset will depend on the config passed
+    and the arguments passed.
+
+    If no session is passed then there won't be a session level folder
+    in the output.
+
+    nested layouts will have subfolders for each datatype
```

Comment:
It would be nice to have the option of having a "session" level too, either as top-level folder (per current Nipoppy expectations for DICOMs) or second-level folder.

Reply:
Currently you can generate dataset with:

- subject/session/datatype/file
- subject/session/all_files
- subject/datatype/files
- subject/all_files

Just to make sure, what other "layout" would be useful?

Reply:
I was thinking of `session/subject/datatype/file` or `session/subject/all_files` because we support that for DICOMs. Though not sure we will actually see this case with non-BIDS Nifti datasets